crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -1603,4 +1603,23 @@ not_a_directory
             ModuleResolutionPathBuf::editable_installation_root(db.system(), "/src").unwrap()
         )));
     }
+
+    #[test]
+    fn resolution_query_invalidated_when_package_deleted() {
+        const SRC: &[FileSpec] = &[("foo/__init__.py", ""), ("foo/bar.py", "")];
+
+        let TestCase { mut db, src, .. } = TestCaseBuilder::new().with_src_files(SRC).build();
+        let memory_fs = db.memory_file_system();
+
+        let foo_bar_name = ModuleName::new_static("foo.bar").unwrap();
+
+        let foo_module = resolve_module(&db, foo_bar_name.clone()).unwrap();
+        assert_eq!(foo_module.file().path(&db), &src.join("foo/bar.py"));
+
+        memory_fs.remove_file(src.join("foo/__init__.py")).unwrap();
+        memory_fs.remove_file(src.join("foo/bar.py")).unwrap();
+        memory_fs.remove_directory(src.join("foo")).unwrap();
```

Comment:
Do we have to delete these files as well? If so, should we call `touch` for them?

Reply:
Argh, you're right. This test doesn't exercise anything at all right now :( It passes on `main` if I `touch` both files.

Reply:
I removed the test, since your file-watching PR adds tests that fail reliably without this fix